agent/lib/kontena/workers/volumes/volume_manager.rb

```diff
@@ -94,16 +98,34 @@ def sync_volume_to_master(docker_volume)
       rpc_client.async.request('/node_volumes/set_state', [node.id, volume])
     end
 
-    def volume_exist?(volume_name)
+    # Checks if given volume exists with the expected driver
+    #
+    # @param [String] name of the volume
+    # @param [String] driver to expect on the volume if already existing
+    # @raise [DriverMismatchError] If the volume is found but using a different driver than expected
+    def __volume_exist?(volume_name, driver)
       begin
         debug "volume #{volume_name} exists"
-        true if Docker::Volume.get(volume_name)
+        volume = Docker::Volume.get(volume_name)
+        if volume && volume.info['Driver'] == driver
+          return true
+        elsif volume && volume.info['Driver'] != driver
+          raise DriverMismatchError.new("Volume driver not as expected. Expected #{driver}, existing volume has #{volume.info['Driver']}")
+        end
       rescue Docker::Error::NotFoundError
         debug "volume #{volume_name} does NOT exist"
         false
       end
     end
 
+    def volume_exist?(volume_name, driver)
```

Comment:
This is the non-crashing wrapper that does `Celluloid::abort` instead of crashing when called as a remote method... the internal `__volumes_exist?` is also called locally within this actor, and it should raise.

How does `Celluloid::abort` behave if called locally by the same actor? If it just raises, then I guess we don't need a wrapper function for this?